diff --git build/build.sh build/build.sh
index 9de796358..d38e10f6a 100755
--- build/build.sh
+++ build/build.sh
@@ -41,29 +41,32 @@ echo "Building targets for ${ARCH}, generated targets in ${TARGETS_DIR} director
 echo "Building ${PKG}/cmd/nginx"
 
 ${GO_BUILD_CMD} \
-  -trimpath -ldflags="-buildid= -w -s \
+  -trimpath=false -ldflags="-buildid= -w -s \
   -X ${PKG}/version.RELEASE=${TAG} \
   -X ${PKG}/version.COMMIT=${COMMIT_SHA} \
-  -X ${PKG}/version.REPO=${REPO_INFO}" \
+  -X ${PKG}/version.REPO=${REPO_INFO} \
+  -X main.version=${TAG}" \
   -buildvcs=false \
   -o "${TARGETS_DIR}/nginx-ingress-controller" "${PKG}/cmd/nginx"
 
 echo "Building ${PKG}/cmd/dbg"
 
 ${GO_BUILD_CMD} \
-  -trimpath -ldflags="-buildid= -w -s \
+  -trimpath=false -ldflags="-buildid= -w -s \
   -X ${PKG}/version.RELEASE=${TAG} \
   -X ${PKG}/version.COMMIT=${COMMIT_SHA} \
-  -X ${PKG}/version.REPO=${REPO_INFO}" \
+  -X ${PKG}/version.REPO=${REPO_INFO} \
+  -X main.version=${TAG}" \
   -buildvcs=false \
   -o "${TARGETS_DIR}/dbg" "${PKG}/cmd/dbg"
 
 echo "Building ${PKG}/cmd/waitshutdown"
 
 ${GO_BUILD_CMD} \
-  -trimpath -ldflags="-buildid= -w -s \
+  -trimpath=false -ldflags="-buildid= -w -s \
   -X ${PKG}/version.RELEASE=${TAG} \
   -X ${PKG}/version.COMMIT=${COMMIT_SHA} \
-  -X ${PKG}/version.REPO=${REPO_INFO}" \
+  -X ${PKG}/version.REPO=${REPO_INFO} \
+  -X main.version=${TAG}" \
   -buildvcs=false \
   -o "${TARGETS_DIR}/wait-shutdown" "${PKG}/cmd/waitshutdown"
